src/security/ZeroSensitiveMemory.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_SECURITY_ZEROSENSITIVEMEMORY_H
+#define SQUID_SRC_SECURITY_ZEROSENSITIVEMEMORY_H
+
+#if HAVE_LIMITS_H
+#include <limits.h>
+#endif
+
+#if HAVE_STRING_H
+#include <string.h>
+#endif
+
+#include <stdexcept>
+
+namespace Security {
+
+inline static void
+ZeroSensitiveMemory(void *dst, const size_t len)
+{
+    if (!dst)
+        throw std::runtime_error("Cannot clear a null buffer");
```

Comment:
Squid usually throws TextExceptions. They may give more information to the admin and improve logging. If possible, please throw a TextException instead.

```suggestion
        throw TextException("attempt to clear a non-empty buffer via a nil pointer", Here());
```

Reply:
I am marking this change request as addressed because the problem it was describing is gone now. I will submit a new change request about the new problem created in the same place.